src/script/event/processor/FederationEventProcessor/FederationEventProcessor.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {FEDERATION_EVENT} from '@wireapp/api-client/lib/event';
+import type {QualifiedId} from '@wireapp/api-client/lib/user';
+import {container} from 'tsyringe';
+import {debounce} from 'underscore';
+
+import {ConversationState} from 'src/script/conversation/ConversationState';
+import {ConversationStatus} from 'src/script/conversation/ConversationStatus';
+import {EventBuilder} from 'src/script/conversation/EventBuilder';
+import {Conversation} from 'src/script/entity/Conversation';
+import {User} from 'src/script/entity/User';
+import {ServerTimeHandler} from 'src/script/time/serverTimeHandler';
+
+import {
+  getUsersToDeleteFromFederatedConversations,
+  getFederationDeleteEventUpdates,
+} from './ConversationFederationUtils';
+
+import {EventProcessor, IncomingEvent} from '../../EventProcessor';
+import {EventRepository} from '../../EventRepository';
+
+export class FederationEventProcessor implements EventProcessor {
```

Comment:
This file is basically a copy/paste of the code removed from the ConversationRepository